src/flamegraph/flamegraph.js

```diff
@@ -123,17 +164,18 @@ function g_to_func(e) {
 function update_text(e) {
     var r = find_child(e, "rect");
     var t = find_child(e, "text");
-    var w = parseFloat(r.attributes.width.value) -3;
-    var txt = find_child(e, "title").textContent.replace(/\\([^(]*\\)\$/,"");
-    t.attributes.x.value = parseFloat(r.attributes.x.value) + 3;
+    var framesWidth = frames.width.baseVal.value;
```

Comment:
What is this `baseVal` business, and why isn't this just `frames.attributes.width.value`?

Reply:
`<svg>` elements have `width` and `height` properties that that are objects with the `SVGAnimatedLength` interface. The `baseVal` will give the value before animation. For the top level `<svg>` we use this because we remove the `width` attribute, but in this case `frames` is the inner `<svg>` and it does have a width attribute. I agree that `frames.attributes.width.value` would be better here.